internal/streamingnode/server/wal/interceptors/shard/utils/stats.go

```diff
@@ -98,3 +101,33 @@ func (s *SegmentStats) Copy() *SegmentStats {
 	s2 := *s
 	return &s2
 }
+
+// InsertMetrics is the metrics of insert operation.
+type InsertMetrics struct {
+	Rows       uint64
+	BinarySize uint64
+}
+
+// Collect collects other metrics.
+func (m *InsertMetrics) Collect(other InsertMetrics) {
+	m.Rows += other.Rows
+	m.BinarySize += other.BinarySize
+}
+
+// Subtract subtract by other metrics.
+func (m *InsertMetrics) Subtract(other InsertMetrics) {
+	if m.Rows < other.Rows {
+		panic("rows cannot be less than zero")
```

Comment:
Is it necessary to enforce strict panic measures here? Can the monitoring system also detect such anomalies if negative numbers appear?